redux/reducers.js

```diff
@@ -0,0 +1,60 @@
+import { combineReducers } from 'redux';
+import * as types from "./actionTypes";
+
+/* Current shape of state:
+ {
+  user: "David",
+  currentChat: id 
+  chats: [ { id: 1,
+    name: 'Tea Chats',
+    users: ['David', 'Nick'] },
+    { id: 2,
+      name: 'Coffee Chats',
+      users: ['David', 'Ryan'] } ]
+  messages: [
+    { id: 0, chats_id: 1, user: 'David', message: 'ayyyyy', emojis: { David: ':joy:', Nick: ':laughing:' } },
+    { id: 1, chats_id: 1, user: 'Nick', message: 'lmao', emojis: {} }
+  ]
+} */
+
+const messageInitialState = {
+  messages: []
+};
+
+const userInitialState = {
+  user: '',
+  currentChat: 0,
+  chats: []
+};
+
+function messageReducer(state = messageInitialState, action) {  
+  console.log("messageReducer");
+  console.log(state); 
+    switch (action.type) {
+    case types.ADD_MESSAGE:
+      return {...state, messages: state.messages.concat(action.payload) };
+    default:
+       return state;
+  }
+}
+
+function userReducer(state = userInitialState, action) {
+  console.log("userReducer");
+  console.log(state); 
+  switch (action.type) {
```

Comment:
another tip i was going to mention before and forgot... you can destructure variables so it saves you time writing!

usually i'll do something like `const { payload, type } = action;`

and then in the function you can just refer to it as "payload" and "type" instead of "action.payload" and "action.type"

Reply:
^💯

Reply:
Nice! That's kind of trippy, haha.